Remotion/Web/ClientScript/Scripts/SmartPage.ts

```diff
@@ -464,13 +470,15 @@ class SmartPage_Context
   // Determines whether the page was loaded from cache.
   public CheckIfCached(): void
   {
-    var field = this._theForm.SmartPage_CacheDetectionField;
-    if (field.value == this._cacheStateHasSubmitted)
+    const pageToken = document.getElementById('SmartPage_InstanceToken')!.getAttribute('value');
+    const hasDifferentPageToken = history.state?.pageToken && history.state.pageToken !== pageToken;
```

Comment:
should we add a comment for the need/semantics of the pagetoken? I think this is a case where the need is only discoverable via trial and error / reverse engineering via debugger. thoughts?